### PR TITLE
Scheduled Updates: Include name param and update schema

### PR DESCRIPTION
**Pull request by obenland**
## Proposed changes:

* Accept name parameter to store alongside plugins.
* Update response schema to include information needed to display schedules in Calypso.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

TBD

### CODE REVIEW

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php [hunk 1]
```diff
@@ -184,7 +198,10 @@ public function create_item( $request ) {
...
 
 		$schedules   = get_option( 'jetpack_update_schedules', array() );
-		$schedules[] = $plugins;
+		$schedules[] = array(
+			'name'    => $request['name'],
+			'plugins' => $plugins,
+		);
 		update_option( 'jetpack_update_schedules', $schedules );
 
 		return rest_ensure_response( $this->generate_schedule_id( $plugins ) );
```
Comment (vishnugopal) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php:207]: It'd be nice if we can store the `id` in `jetpack_update_schedules` too and return it in `get_item` and `get_items`. I found it a bit strange that we return an id for create, but not for get or list.

Comment (bogiii) [reply projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php:207]: +1, without ID... I can not delete or update items.

**Comment by ouikhuan** [comment]
Some thoughts about the `generate_schedule_id` function, we might want to make sure the plugins come in the same order every single time and sort it. For example, the result of the following will be different:
```
Input:
$plugins = array(
	'gutenberg/gutenberg.php',
	'custom-plugin/custom-plugin.php',
);
Output: 0e8084c8c932e72b700da80f81c153ff
```

```
Input:
$plugins = array(
	'custom-plugin/custom-plugin.php',
	'gutenberg/gutenberg.php',
);
Output: 6e5d5fed314ed0b81b54723ef1489c29
```
But they have the same plugins checked.

**Comment by obenland** [comment]
@Automattic/caribou Could you see if https://github.com/Automattic/jetpack/pull/36071 addresses some of the issues you mentioned?

It sorts plugin arrays, gets rid of the `jetpack_update_schedules` option entirely, and includes the hash id in `get_items()` responses. It's quite a bit more elegant, I think.

**Comment by obenland** [comment]
Closing in favor of https://github.com/Automattic/jetpack/pull/36071

**Pull request closed without merging**